### PR TITLE
feat: implement DB read replicas

### LINKED ISSUE #286: Use read replicas
https://github.com/gajus/slonik#routing-queries-to-different-connections

### PR DESCRIPTION
**Pull request by dialexo**
Closes #286

### CODE REVIEW

File: src/plugins/database.ts [hunk 1]
```diff
@@ -23,9 +27,45 @@ const plugin: FastifyPluginAsync<DatabasePluginOptions> = async (fastify, { uri,
...
   }
 
+  // read replicas load balancing
+  // see https://github.com/gajus/slonik#routing-queries-to-different-connections
+  if (readReplicaUris) {
+    const readOnlyPools = readReplicaUris.map((uri) => createPool(uri, options));
+
+    const readOnlyInterceptor: Interceptor = {
+      beforePoolConnection: (connectionContext) => {
+        if (!connectionContext.query?.sql.trim().startsWith('SELECT ')) {
```
Comment (pyphilia) [inline src/plugins/database.ts:42]:
```suggestion
(!connectionContext.query?.sql?.trim()?.startsWith?.('SELECT ')) {
```
Or more if?

File: src/plugins/database.ts [hunk 2]
```diff
@@ -23,9 +27,45 @@ const plugin: FastifyPluginAsync<DatabasePluginOptions> = async (fastify, { uri,
...
+          return null;
+        }
+
+        // This is a convention for the edge-cases where a SELECT query includes a volatile function.
+        // Adding a @volatile comment anywhere into the query bypasses the read-only route, e.g.
+        // sql.unsafe`
+        //   # @volatile
+        //   SELECT write_log()
+        // `
+        if (connectionContext.query?.sql.includes('@volatile')) {
```
Comment (pyphilia) [inline src/plugins/database.ts:53]: Save as above?

**Review by pyphilia — APPROVED** [review]
LGTM 🥾

**Comment by dialexo** [comment]
TODO: create corresponding secrets

**Pull request merged**